src/web/app/components/sessions-table/submission-status-tooltip.pipe.ts

```diff
@@ -47,9 +49,30 @@ export class SubmissionStatusTooltipPipe implements PipeTransform {
       default:
     }
 
+    if (deadlines && this.hasOngoingExtension(deadlines)) {
+      msg += ', with current ongoing session extensions';
+    }
+
     msg += '.';
 
     return msg;
   }
 
+  private hasOngoingExtension(deadlines: {
+    studentDeadlines: Record<string, number>,
+    instructorDeadlines: Record<string, number>
+  }): boolean {
+    const timeNow = Date.now();
+
+    for (const studentEmail in deadlines.studentDeadlines) {
+      if (deadlines.studentDeadlines[studentEmail] > timeNow) return true;
+    }
+
+    for (const instructorDeadline in deadlines.instructorDeadlines) {
+      if (deadlines.studentDeadlines[instructorDeadline] > timeNow) return true;
+    }
+
+    return false;
+  }
```

Comment:
Looks like a duplicated code in another file? Is there tear place to put it?

Reply:
I'll refactor the pipe into the teammatescommon module, and put the helper method into a separate class